x-pack/legacy/plugins/apm/public/context/UrlParamsContext/index.tsx

```diff
@@ -26,12 +30,29 @@ interface TimeRange {
 function useUiFilters({
   kuery,
   environment,
-  transactionType
-}: IUrlParams): UIFilters {
-  return useMemo(() => ({ kuery, environment, transactionType }), [
+  transactionType,
+  ...localFilters
+}: Pick<
+  IUrlParams,
+  'kuery' | 'environment' | 'transactionType' | LocalFilterName
+>): UIFilters {
+  return useMemo(() => {
+    const mappedLocalFilters = mapValues(
+      pick(localFilters, filterNames) as Record<
+        LocalFilterName,
+        IUrlParams[LocalFilterName]
+      >,
+      val => (val ? val.split(',') : [])
```

Comment:
I don't think I understand what's going on here. Why the `...localFilters` above instead of writing the filters out explicitly?

Reply:
I'm trying to prevent writing out those filter names in various places like we do with other params. It's fully covered by TypeScript as you can see in the screenshot:

![image](https://user-images.githubusercontent.com/352732/61589367-2334db00-aba9-11e9-863b-b3b0b5ed8b0b.png)

Multiple values are formatted to a query parameter as a comma-separated list, so that's mapped back over here before it's passed to the useLocalFilters hook. I don't think I need the `pick` here, if you pass in an unknown local filter TypeScript will complain about it anyway.

Do you have any suggestions on making this more clear?